docs/topics/sync-vs-async.rst

```diff
@@ -101,7 +101,7 @@ When to use a synchronous function
 ----------------------------------
 
 As an inverse of the previous paragraph, it follows that synchronous functions should
-be used for non-blocking, non-computationally intensive tasks. The synchronous execution
+be used for non-io intensive tasks. The synchronous execution
```

Comment:
This makes it sound like you should use synchronous functions for lightweight IO tasks, which is the opposite of what the paragraph is trying to say